### PR TITLE
Update conventional plane stress formulation in peridynamics module

### LINKED ISSUE #16652: Correct the stiffness tensor for conventional plane stress formulation in peridynamics test
## Bug Description

One component of the stiffness tensor for conventional plane stress formulation was implemented wrong, which only affects the shear stress calculation.
## Impact

This conventional formulation was kept in the test directory only for the purpose of having an option to compare the weak formulation with the conventional formulation. Only one test uses this formulation.

### PR DESCRIPTION
**Pull request by hchen139**
closes #16652

### CODE REVIEW

**Comment by hchen139** [comment]
@bwspenc One component of the stiffness tensor for the conventional plane stress formulation was wrongly set to zero. I have made the correction and updated some other input parameter requirements for a PD auxkernel that postprocess the stress tensor for BPD and OSPD models.

File: modules/peridynamics/src/auxkernels/NodalRankTwoPD.C [hunk 1]
```diff
@@ -85,23 +80,36 @@ NodalRankTwoPD::NodalRankTwoPD(const InputParameters & parameters)
   for (unsigned int i = 0; i < coupledComponents("displacements"); ++i)
     _disp_var.push_back(getVar("displacements", i));
 
+  _Cijkl.zero();
   if (_rank_two_tensor == "stress" && !isParamValid("youngs_modulus") &&
       !isParamValid("poissons_ratio"))
     mooseError("Both Young's modulus and Poisson's ratio must be provided for stress calculation");
+  else if (_rank_two_tensor == "stress")
+  {
+    _youngs_modulus = getParam<Real>("youngs_modulus");
```
Comment (bwspenc) [inline modules/peridynamics/src/auxkernels/NodalRankTwoPD.C:89]:
```
if (_rank_two_tensor == "stress")
{
  check for defined parameters
  ...
}

File: modules/peridynamics/src/auxkernels/NodalRankTwoPD.C [hunk 2]
```diff
@@ -85,23 +80,36 @@ NodalRankTwoPD::NodalRankTwoPD(const InputParameters & parameters)
...
 
   if (_has_temp && !isParamValid("thermal_expansion_coeff"))
     mooseError("thermal_expansion_coeff is required when temperature is coupled");
+  else if (_has_temp)
+    _alpha = getParam<Real>("thermal_expansion_coeff");
 
   if (_output_type == "component" && !(isParamValid("index_i") || isParamValid("index_j")))
     mooseError("The output_type is 'component', but no 'index_i' and 'index_j' are provided!");
+  else if (_output_type == "component")
+  {
```
Comment (bwspenc) [inline modules/peridynamics/src/auxkernels/NodalRankTwoPD.C:109]: Same rearrangement of logic as above

**Review by bwspenc — CHANGES_REQUESTED** [review]

File: modules/peridynamics/src/auxkernels/NodalRankTwoPD.C [hunk 3]
```diff
@@ -85,23 +80,42 @@ NodalRankTwoPD::NodalRankTwoPD(const InputParameters & parameters)
...
-  if (_rank_two_tensor == "stress" && !isParamValid("youngs_modulus") &&
-      !isParamValid("poissons_ratio"))
-    mooseError("Both Young's modulus and Poisson's ratio must be provided for stress calculation");
+  _Cijkl.zero();
+  if (_rank_two_tensor == "stress")
+  {
+    if (!isParamValid("youngs_modulus") || !isParamValid("poissons_ratio"))
+      mooseError(
+          "Both Young's modulus and Poisson's ratio must be provided for stress calculation");
+    else
```
Comment (bwspenc) [inline modules/peridynamics/src/auxkernels/NodalRankTwoPD.C:89]: You don't actually need to put the following lines of code inside the `else` block, but it doesn't hurt anything, so I'll let that slide.

**Review by bwspenc — APPROVED** [review]

**Pull request merged**